context/src/jmh/java/io/opentelemetry/context/ContextBenchmark.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.context;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+/** Benchmark for Context. */
+@Threads(value = 1)
+@Fork(3)
+@Warmup(iterations = 10, time = 1)
+@Measurement(iterations = 5, time = 1)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+public class ContextBenchmark {
+
+  @Param({"2", "3", "4", "5", "10", "20", "40"})
+  private int size;
+
+  private List<ContextKey<String>> keys;
+  private Context context = Context.root();
+
+  @Setup
+  public void setup() {
+    keys = new ArrayList<>();
+    for (int i = 0; i < size; i++) {
+      ContextKey<String> key = ContextKey.named(Integer.toString(i));
+      context = context.with(key, "value");
+      keys.add(key);
+    }
+  }
+
+  /** Read value for first key. */
+  @Benchmark
+  public String readFirst() {
+    return context.get(keys.get(0));
+  }
+
```

Comment:
Would it a get to the middle element make sense?

Reply:
For trie getting any element should have almost the same impact.

Reply:
So yes a test for the middle should be acceptable

Reply:
Seems interesting to have, will try to run it later. Thanks for the suggestion!